### PR TITLE
UIIN-827: Fix empty fields for lists

### PR DESCRIPTION
**Pull request by Vladyslav-Velytskyi**

### CODE REVIEW

File: src/ViewHoldingsRecord.js [hunk 1]
```diff
@@ -460,51 +469,87 @@ class ViewHoldingsRecord extends React.Component {
...
+                'Staff only': <FormattedMessage id="ui-inventory.staffOnly" />,
+                'Note': notesList === emptyList ? <FormattedMessage id="ui-inventory.note" /> : noteName,
+              }}
+              columnWidths={{
+                'Staff only': '25%',
+                'Note': '75%',
+              }}
+              formatter={
+                {
+                  'Staff only': x => (notesList === emptyList ? noValue : formatterStaffOnly(x)),
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.js]: Do you need parentheses here?
```suggestion
                  'Staff only': x => notesList === emptyList ? noValue : formatterStaffOnly(x),
```

Comment (Vladyslav-Velytskyi) [reply src/ViewHoldingsRecord.js]: LintError, in function which returns ternary condition should be in brackets

File: src/ViewHoldingsRecord.js [hunk 2]
```diff
@@ -806,36 +830,34 @@ class ViewHoldingsRecord extends React.Component {
...
+                      columnWidths={{
+                        'URL relationship': '16%',
+                        'URI': '16%',
+                        'Link text': '16%',
+                        'Materials specified': '16%',
+                        'URL public note': '32%',
+                      }}
+                      formatter={{
+                        'URL relationship': x => this.refLookup(referenceTables.electronicAccessRelationships, get(x, ['relationshipId'])).name || noValue,
+                        'URI': x => (get(x, ['uri']) ? <a href={get(x, ['uri'])} style={wrappingCell}>{get(x, ['uri'])}</a> : noValue),
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.js]: Consider splitting it into separate lines

File: src/ViewInstance.js [hunk 3]
```diff
@@ -592,28 +619,40 @@ class ViewInstance extends React.Component {
...
+              columnWidths={{
+                'Staff only': '25%',
+                'Note': '75%',
+              }}
+              formatter={
+                {
+                  'Staff only': x => (notesList === emptyList ? noValue : formatterStaffOnly(x)),
+                  'Note': x => get(x, ['note']) || noValue,
+                }
+              }
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js]: I think it is not necessary to wrap brackets on a new line

File: src/ViewInstance.js [hunk 4]
```diff
@@ -713,26 +749,44 @@ class ViewInstance extends React.Component {
...
-        <FormattedDate value={checkIfElementIsEmpty(instanceData.instanceStatusUpdatedDate)} />
-        <br />
-        <FormattedTime value={checkIfElementIsEmpty(instanceData.instanceStatusUpdatedDate)} />
-      </React.Fragment>
-    );
+    const formattedStatusUpdatedDate = instanceData.instanceStatusUpdatedDate !== '-'
+      ? (
+        <Fragment>
+          <FormattedDate value={instanceData.instanceStatusUpdatedDate} />
+          <br />
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js]: Consider wrapping value to block element instead of using <br> tag

File: src/utils.js [hunk 5]
```diff
@@ -145,13 +147,15 @@ export const validateAlphaNumericField = value => {
 
 export const checkIfElementIsEmpty = element => (element === '-' ? noValue : element);
 
+export const checkIfArrayIsEmpty = array => (!isEmpty(array) ? array : emptyList);
+
 export const convertArrayToBlocks = elements => (!isEmpty(elements)
-  ? elements.map((line, i) => <div key={i}>{line}</div>)
-  : '-');
+  ? elements.map((line, i) => (line !== null ? <div key={i}>{line}</div> : noValue))
```
Comment (mariia-aloshyna) [inline src/utils.js]: Maybe it makes sense to check just line? If the line will be equal to undefined or empty value, this check will not pass. 
```suggestion
  ? elements.map((line, i) => (line ? <div key={i}>{line}</div> : noValue))
```

File: src/utils.js [hunk 6]
```diff
@@ -168,3 +174,22 @@ export const callNumberLabel = holdingsRecord => {
 
   return parts.join(' ');
 };
+
+export const formatterStaffOnly = x => (get(x, ['staffOnly'])
```
Comment (mariia-aloshyna) [inline src/utils.js]: Rename it please to "staffOnlyFormatter"

**Pull request merged**